### PR TITLE
Documentation backbone for Android SDK

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation
 
Provide backbone for documentation - same format as other documentation.
Align documentation into procedures.
Reformat existing content into guides (more practical info for developers)
Provide documentation for core (minimal one that can be extended over the time).

## Progress

- [x] Adjust structure to specific package
- [x] Write getting started guide
- [x] Write contributor guide

### CODE REVIEW

File: docs/contributing-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
+== AeroGear Service Developer Guide
+
+Following guide describes architecture for IOS SDK and instructions for developing new AeroGear Services.
```
Comment (aidenkeating) [inline docs/contributing-guide.adoc]: IOS SDK -> Android SDK

File: docs/contributing-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+For instructions about using SDK in user applications please follow link:./getting-started.adoc[User Getting Started Guide]
+
+image:./images/diagram.svg[diagram,450,450,role="right"]
```
Comment (aidenkeating) [inline docs/contributing-guide.adoc]: This diagram file will need to be updated to reflect Android-y stuff. I can do this if needed in a separate PR.

File: docs/contributing-guide.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+=== SDK architecture
+
+SDK aggregates individual services SDK into one single manageable namespace. Services like AeroGear Sync, Push are build as separate SDK's that can be included into any existing or new IOS applications. SDK using gradle for dependency resolution and distribution. Each SDK is represented as separate gradle module. Modules are used and are tested in `example` application that is also used for building and demo purposes.
```
Comment (aidenkeating) [inline docs/contributing-guide.adoc]: new IOS applications -> new Android applications

File: README.md [hunk 4]
```diff
@@ -14,17 +14,44 @@
...
 
 If you would like to help develop AeroGear you can join our [developer's mailing list](https://groups.google.com/forum/#!forum/aerogear), join #aerogear on Freenode, or shout at us on Twitter @aerogears.
 
 Also takes some time and skim the [contributor guide](CONTRIBUTING.md)
 
+## Contributing
+
+See [General Contributing Guide](./CONTRIBUTING.md)
+
+## License
```
Comment (wei-lee) [inline README.md]: looks like this section and the above section are duplicated?

File: docs/contributing-guide.adoc (hunk 2 above)
Comment (wtrocki) [reply docs/contributing-guide.adoc]: I could update it. Can you give me some example what will be the best to get that in?

File: docs/contributing-guide.adoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+Following guide describes architecture for Android SDK and instructions for developing new AeroGear Services.
+
+For instructions about using SDK in user applications please follow link:./getting-started.adoc[User Getting Started Guide]
+
+image:./images/diagram.svg[diagram,450,450,role="right"]
+
+=== SDK architecture
+
+SDK aggregates individual services SDK into one single manageable namespace. Services like AeroGear Sync, Push are build as separate SDK's that can be included into any existing or new Android applications. SDK using gradle for dependency resolution and distribution. Each SDK is represented as separate gradle module. Modules are used and are tested in `example` application that is also used for building and demo purposes.
```
Comment (wei-lee) [inline docs/contributing-guide.adoc]: we don't really have Sync and Push at the moment, so I think mention them here will cause some confusion. Better replace it with the services we currently have, like auth and metrics.

File: docs/contributing-guide.adoc (hunk 2 above)
Comment (aidenkeating) [reply docs/contributing-guide.adoc]: Probably something like [Auth] and [Metrics], replacing "Swift source code" and "Podspec file". Everything else seems accurate.

File: README.md (hunk 4 above)
Comment (wtrocki) [reply README.md]: Done

**Pull request merged**